packages/components/src/internal/components/domainproperties/actions.ts

```diff
@@ -418,30 +418,108 @@ export function getMaxPhiLevel(containerPath?: string): Promise<string> {
     });
 }
 
-export function parseCalculatedColumn(
+// exported for jest testing
+export function getCastStatement(key: string, type: string): string {
+    switch (type) {
+        case 'INTEGER':
+        case 'SAMPLE':
+        case 'USERS':
+            return `CAST(1 AS INTEGER) AS "${key}"`;
+        case 'DOUBLE':
+        case 'DECIMAL (FLOATING POINT)':
+        case 'VISITID':
+            return `CAST(1.1 AS DOUBLE) AS "${key}"`;
+        case 'BOOLEAN':
+            return `CAST(TRUE AS BOOLEAN) AS "${key}"`;
+        case 'DATETIME':
+        case 'VISITDATE':
+            return `CAST(CURDATE() AS TIMESTAMP) AS "${key}"`;
+        case 'DATE':
+            return `CAST(CURDATE() AS DATE) AS "${key}"`;
+        case 'TIME':
+            return `CAST('13:00' AS TIME) AS "${key}"`;
+        default:
+            return `CAST('Testing' AS VARCHAR) AS "${key}"`;
+    }
+}
+
+export async function parseCalculatedColumn(
     expression: string,
     columnMap: Record<string, string>,
+    phiColumns: string[],
     containerPath?: string
 ): Promise<{ error: string; type: string }> {
-    return new Promise((resolve, reject) => {
-        if (!expression || expression?.trim()?.length === 0) {
-            resolve({ error: 'Error: an expression value is required.', type: undefined });
-            return;
-        }
+    if (!expression || expression?.trim()?.length === 0) {
+        return { error: 'Error: an expression value is required.', type: undefined };
+    }
 
+    try {
+        const sql =
+            'SELECT\n' +
+            expression +
+            '\nFROM (SELECT ' +
+            Object.keys(columnMap)
+                .map(key => {
+                    return getCastStatement(key, columnMap[key]);
+                })
+                .join(',\n') +
+            ') AS x' +
+            ' WHERE 1=0';
+
+        await selectRowsDeprecated({
+            schemaName: 'core',
+            sql,
+            includeTotalCount: false,
+        });
+    } catch (e) {
+        const error = resolveErrorMessage(e, 'data', undefined, undefined, undefined, true);
+        return { error: _sanitizeParseCalculatedColumnError(error), type: undefined };
```

Comment:
You might change `_sanitizeParseCalculatedColumnError` to include the call to `resolveErrorMessage`, just so it is all encompassed in one place.

Reply:
done